modules/ml/src/main/java/org/apache/ignite/ml/tree/DecisionTree.java

```diff
@@ -62,21 +65,22 @@
      * @param decisionTreeLeafBuilder Decision tree leaf builder.
      */
     DecisionTree(int maxDeep, double minImpurityDecrease, StepFunctionCompressor<T> compressor,
-        DecisionTreeLeafBuilder decisionTreeLeafBuilder) {
+        DecisionTreeLeafBuilder decisionTreeLeafBuilder, boolean useIndex) {
```

Comment:
I think it shouldn't be a constructor parameter. This is a not a part of algorithm user settings. This is a performance improvement. Design it via setter or something else. And set the default value (true, for example). Or provide two the same realization like DT and DT on steroids.

Reply:
I like constructor to be honest. Would be great ho have constructor with default value as well.

Reply:
okay...
I prefer Builder-like variant

Reply:
The story with constructors make a problem for injecting new values into trainer fields.
Of course we can pass parameters via constructors and duplicates with setters, but it would be 2-approach problem for our users.